### PR TITLE
set okteto context and namespace from env vars

### PR DESCRIPTION
**Pull request by andreafalzetti**
# Proposed changes

Fixes DEV-281

## How to validate

You need at least 2 different contexts to validate this change. 

### Scenario 1 - no context

Without any current context, or `--context`, or `OKTETO_CONTEXT` the CLI should tell you that you have no context.

Try the following:

```
OKTETO_HOME=$(mktemp -d) okteto endpoints
# or
OKTETO_HOME=$(mktemp -d) okteto context show
```

And validate that you get a message in both cases saying your context is not set.

### Scenario 2 - current context

Assuming you have a current context:

1. run `okteto context show`
2. validate that the context is correct

### Scenario 3 - current context + `--context` flag

The flag should have priority over the current context, to validate this scenario, run:

1. run `okteto context show`
2. observe which context you have as current (let's callit `<CTX1>`)
3. Run `okteto endpoints --context <CTX2>`
4. the CLI should say you are using `<CTX2>`
5. check that the namespace is also selected correctly based on `<CTX2>`
6. now run: `okteto endpoints --context <CTX2> --namespace <NS2>`
7. check that the `<NS2>` is picked up as expected

### Scenario 4 - current context + `OKTETO_CONTEXT`

1. run `okteto context show`
2. observe which context you have as current (let's callit `<CTX1>`)
3. Run `OKTETO_CONTEXT=<CTX2> okteto endpoints`
4. the CLI should say you are using `<CTX2>`
5. check that the namespace is also selected correctly based on `<CTX2>`
6. now run: `OKTETO_CONTEXT=<CTX2> OKTETO_NAMESPACE=<NS2> okteto endpoints`
7. check that the `<NS2>` is picked up as expected

### Scenario 5 - current context + `OKTETO_CONTEXT` + `--context`

For this scenario it's ideal to use 3 different contextes to avoid ambiguities.

1. run `okteto context show`
2. observe which context you have as current (let's callit `<CTX1>`)
3. Run `OKTETO_CONTEXT=<CTX2> okteto endpoints --context <CTX3>`
4. the CLI should say you are using `<CTX3>`
5. check that the namespace is also selected correctly based on `<CTX3>`
6. now run: `OKTETO_CONTEXT=<CTX2> OKTETO_NAMESPACE=<NS2> okteto endpoints --context <CTX3> --namespace <NS3>`
7. check that the `<NS3>` is picked up as expected

### Scenario 6 - current context + `.env` with `OKTETO_CONTEXT` + `--context`

1. run `okteto context show`
1. observe which context you have as current (let's callit `<CTX1>`)
1. create `.env` with `OKTETO_CONTEXT=<CTX2>`
1. Run `okteto endpoints --context <CTX3>`
1. the CLI should say you are using `<CTX3>`
1. check that the namespace is also selected correctly based on `<CTX3>`
1. now add `OKTETO_NAMESPACE=<NS2>` to the `.env`
1. Run: `okteto endpoints --context <CTX3> --namespace <NS3>`
1. check that the `<NS3>` is picked up as expected

### Scenario 7 - current context + manifest `context:` + `--context`

1. run `okteto context show`
1. observe which context you have as current (let's callit `<CTX1>`)
1. set `context: <CTX2>` in your manifest
1. Run `okteto endpoints --context <CTX3>`
1. the CLI should say you are using `<CTX2>`
1. check that the namespace is also selected correctly based on `<CTX2>`
1. now add `namespace: <NS2>` in your manifest
1. now run: `okteto endpoints --context <CTX3> --namespace <NS3>`
1. check that the `<NS2>` is picked up as expected

### Scenario 8 - current context + `OKTETO_URL`

1. run `okteto context show`
2. observe which context you have as current (let's callit `<CTX1>`)
3. Run `OKTETO_URL=<CTX2> okteto endpoints`
4. the CLI should say you are using `<CTX2>`
5. check that the namespace is also selected correctly based on `<CTX2>`
6. now run: `OKTETO_URL=<CTX2> OKTETO_NAMESPACE=<NS2> okteto endpoints`
7. check that the `<NS2>` is picked up as expected

### Out of scope scenarios:

These are lower-priority scenarios that may be worked on with a follow-up ticket/PR.

- Scenario 8 - current context + `OKTETO_CONTEXT` var set in Catalog
- Scenario 9 - current context + `OKTETO_CONTEXT` var set in Catalog + `--context`
- Scenario 10 - current context + `OKTETO_CONTEXT` set as `--var` flag in deploy
- Scenario 11 - current context + `OKTETO_CONTEXT` set as User variable
- Scenario 12- current context + `OKTETO_CONTEXT` set as Cluster variable

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

### CODE REVIEW

File: cmd/context/options.go [hunk 1]
```diff
@@ -95,10 +95,11 @@ func (o *Options) InitFromEnvVars() {
 		usedEnvVars = append(usedEnvVars, model.OktetoNamespaceEnvVar)
 	}
 
-	if len(usedEnvVars) == 1 {
-		oktetoLog.Warning("Initializing context with the value of %s environment variable", usedEnvVars[0])
-	} else if len(usedEnvVars) > 1 {
-		oktetoLog.Warning("Initializing context with the value of %s and %s environment variables", strings.Join(usedEnvVars[0:len(usedEnvVars)-1], ", "), usedEnvVars[len(usedEnvVars)-1])
+	if o.Show {
```
Comment (jLopezbarb) [inline cmd/context/options.go:98]: Why are we just showing this warning when the `o.Show` is set? I think this could prevent the user to know which context is using if the command is not a context one

Comment (andreafalzetti) [reply cmd/context/options.go:98]: @jLopezbarb without this, if I run `okteto context show`, the warning appears breaking the JSON or YAML formatting. Do you have an alternative?

**Review by maroshii — COMMENTED** [review]
I don't have enough context in the CLI or understand how context works internally to confidently approve this PR. Deferring approval to @jLopezbarb

**Comment by andreafalzetti** [comment]
Moving to draft until e2e are passing

File: cmd/context/options.go (hunk 1 above)
Comment (andreafalzetti) [reply cmd/context/options.go:98]: good point @ifbyol, I've also noticed that `okteto endpoints --output=json` is affected by this. Will make a change before opening the PR for review again. Thanks both!

**Review by jLopezbarb — APPROVED** [review]
LGTM besides of the e2e tests

**Comment by maroshii** [comment]
This looks like quite a sensitive change for a backport

**Comment by andreafalzetti** [comment]
> This looks like quite a sensitive change for a backport

I agree, as we discussed in our daily this should get in `2.27` but we wanted to start testing, so it was done this way.

A way to help is to review and test all scenarios.

**Comment by andreafalzetti** [comment]
> > A way to help is to review and test all scenarios
> 
> I can do that. I would expect the same amount of testing in my PRs though.
> 
> The PR keeps changing from draft to ready to review and commits keep getting added so please let me know when I can test

In the CLI repo we started adding how to test to facilitate the reviewer exactly for this purpose. When the steps are there, it's easy to try out and even try "breaking" the CLI by coming up with unlisted scenarios (optionally). If the PR doesn't contain any explaination, it's harder to do it. 

> The PR keeps changing from draft to ready to review and commits keep getting added so please let me know when I can test
Thanks to testing it we found scenarios that were not working, so I moved it to draft until CI is passing again. I've pushed new changes now, waiting for the CI to be green and moving to ready again.

**Comment by andreafalzetti** [comment]
> I just don't agree with giving the burden of proof to the reviewer.

The effort I put in writing "How to validate" a PR for me is important for 2 reasons, mainly to make the life of the reviewer easier, second to make sure when we come back to this PR in the future, even a new joiner can understand what the change impacted and how to test it. It also tells you what I've tested and what not. Some scenarios sometimes are ticky to reproduce, so having a sample manifest/repo often it helps.

> I would actually expect the author of the PR to have already done those tests.

The assumption is that the author has done the written validations and they are sharing it for reviewes to also test then and make sure the quality of the CLI is high. So far this helped catch panics, improve UX (i.e. duplicated logs) and catch edge case scenarios. It also means spending more time reviewing, I agree.

> This is very inefficient given that everybody would need to understand EVERY ticket and test all possible scenario to review. 

I think we should aim at build, run and test most changes, personally I believe that it will lead to a better product, but it's my view. Especially for a change sensitive, as you said, I would expect my team to help me test my change and help me catch any possible bug before it gets to a user, I am not perfect, and I can miss something.

**Comment by andreafalzetti** [comment]
Now that I said that, I've realized that the latest changes made this afternoon are passing the E2E but I didn't re-test from scratch and noticed a bad behavioir in some scenarios :(

**Review by jLopezbarb — APPROVED** [review]
